File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int s = 0;
    public TreeNode bstToGst(TreeNode root) {
        dfs(root);
        return root;
    }

    public void dfs(TreeNode t){
        if(t==null) return ;
        dfs(t.right);
        s += t.val;
        t.val = s;
        dfs(t.left);
        
    }

}

File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int s =0;
    void dfs(TreeNode* t){
        if(t==NULL) return;
        dfs(t->right);
        s += t->val;
        t->val = s;
        dfs(t->left);
        
    }
    TreeNode* bstToGst(TreeNode* root) {
        dfs(root);
        return root;
    }
};